platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/PlatformStateService.java

```diff
@@ -54,21 +63,53 @@ public void registerSchemas(@NonNull final SchemaRegistry registry) {
         SCHEMAS.forEach(registry::register);
     }
 
+    /**
+     * Sets the active roster to the given roster.
+     * @param roster the roster to set as active
+     */
+    public void setActiveRosterFn(@NonNull final Supplier<Roster> roster) {
+        ACTIVE_ROSTER.set(requireNonNull(roster));
+    }
+
+    /**
+     * Clears the active roster.
+     */
+    public void clearActiveRosterFn() {
+        ACTIVE_ROSTER.set(null);
+    }
+
     /**
      * Given a {@link MerkleStateRoot}, returns the creation version of the platform state if it exists.
      * @param root the root to extract the creation version from
      * @return the creation version of the platform state, or null if the state is a genesis state
      */
-    @SuppressWarnings("unchecked")
     public SemanticVersion creationVersionOf(@NonNull final MerkleStateRoot root) {
         requireNonNull(root);
-        if (root.getNumberOfChildren() == 0) {
-            return null;
-        }
+        final var state = platformStateOf(root);
+        return state == null ? null : state.creationSoftwareVersionOrThrow();
+    }
+
+    /**
+     * Given a {@link MerkleStateRoot}, returns the round number of the platform state if it exists.
+     * @param root the root to extract the round number from
+     * @return the round number of the platform state, or zero if the state is a genesis state
+     */
+    public long roundOf(@NonNull final MerkleStateRoot root) {
+        requireNonNull(root);
+        final var platformState = platformStateOf(root);
+        return platformState == null
+                ? 0L
+                : platformState
+                        .consensusSnapshotOrElse(ConsensusSnapshot.DEFAULT)
+                        .round();
+    }
```

Comment:
This logic is already implemented in `ReadablePlatformStateStore`. And the functionality of getting the round given a `State` (or a `MerkleStateRoot`) has been implemented in the `RosterRetriever` already. I suggest to avoid code duplication. We should either remove that replica and use this one, or not introduce this one and keep using the `RosterRetriever` instead.

Minor: all the `...Of()` methods in this class don't seem to access instance variables and hence can all be made `static` to make them usable in other places.

Reply:
`ReadablePlatformStateStore` requires the States API to have been initialized, and this accessor's purpose is to **setup** the code that initializes the States API.

Reply:
I see. Makes sense.